firmware/ec/.clang-format

```diff
@@ -0,0 +1,108 @@
+# clang-format configuration file. Intended for clang-format >= 4.
+#
+# Modified from linux kernel .clang-format file. 
+#
+# For more information, see:
+#
+#   Documentation/process/clang-format.rst
+#   https://clang.llvm.org/docs/ClangFormat.html
+#   https://clang.llvm.org/docs/ClangFormatStyleOptions.html
+#
+---
+AccessModifierOffset: -4
+AlignAfterOpenBracket: Align
+AlignConsecutiveAssignments: false
+AlignConsecutiveDeclarations: false
+AlignEscapedNewlinesLeft: true
+AlignOperands: true
+AlignTrailingComments: false
+AllowAllParametersOfDeclarationOnNextLine: false
+AllowShortBlocksOnASingleLine: false
+AllowShortCaseLabelsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: None
+AllowShortIfStatementsOnASingleLine: false
+AllowShortLoopsOnASingleLine: false
+AlwaysBreakAfterDefinitionReturnType: None
+AlwaysBreakAfterReturnType: None
```

Comment:
I think All wouldn't result in matching the guidelines, either, as this is "The function declaration return type breaking style to use." so it would create breaks after the function return type in the declarations.

Reply:
In common usage of AlwaysBreakAfterReturnType , None seems to be more popular.  Yet in our code base the return is always in a line of its own.  Is this due to BreakBeforeBraces?

Reply:
My understanding of this setting it is for the return **type** not the return **statement**.
i.e. a setting of `All` would create this in the linked guidelines function:
```
static uint32_t 
mapLookup(SampleState state, bool uselessParam)
{
```
Having the BraceWrapping setting for functions that it always breaks might help guarantee that we have a break after the return statement, if I run the linter on the example code it keeps the return statements as is.
I'm going to add the example code from the linked guidelines as .c and .h, and see what changes I need to do so that the linter doesn't modify anything.

Reply:
I see, return type is consistent with this: https://clang.llvm.org/docs/ClangFormatStyleOptions.html